tests/lzma.rs

```diff
@@ -75,30 +100,79 @@ fn round_trip_file(filename: &str) {
 fn decomp_big_file(compfile: &str, plainfile: &str) {
     use std::io::Read;
 
-    let mut expected = Vec::new();
-    std::fs::File::open(plainfile)
-        .unwrap()
-        .read_to_end(&mut expected)
-        .unwrap();
-    let mut f = std::io::BufReader::new(std::fs::File::open(compfile).unwrap());
-    let mut decomp: Vec<u8> = Vec::new();
-    lzma_rs::lzma_decompress(&mut f, &mut decomp).unwrap();
-    assert!(decomp == expected)
+    let expected = {
+        let mut expected = Vec::new();
+        std::fs::File::open(plainfile)
+            .unwrap()
+            .read_to_end(&mut expected)
+            .unwrap();
+        expected
+    };
+
+    // test non-streaming decompression
+    {
+        let input = {
+            let mut input = Vec::new();
+            std::fs::File::open(compfile)
+                .unwrap()
+                .read_to_end(&mut input)
+                .unwrap();
+            input
+        };
+
+        let mut input = std::io::BufReader::new(input.as_slice());
+        let mut decomp: Vec<u8> = Vec::new();
+        lzma_rs::lzma_decompress(&mut input, &mut decomp).unwrap();
+        assert_eq!(decomp, expected);
+    }
+
+    #[cfg(feature = "stream")]
+    // test streaming decompression
+    {
+        let mut compfile = std::fs::File::open(compfile).unwrap();
+        let mut stream = lzma_rs::decompress::Stream::new(Vec::new());
+
+        // read file in chunks
+        let mut tmp = [0u8; 1024];
+        while {
+            let n = compfile.read(&mut tmp).unwrap();
+            stream.write_all(&tmp[0..n]).unwrap();
+
+            n > 0
+        } {}
```

Comment:
This logic is not readable. Can you use `if` together with `break` for the exit condition?